File: PrismaApi/PrismaApi.Domain/Dtos/EdgeDtos.cs
using System;
using System.Text.Json.Serialization;

namespace PrismaApi.Domain.Dtos;

public class EdgeDto
{
    [JsonPropertyName("id")]
    public Guid Id { get; set; } = Guid.NewGuid();
    [JsonPropertyName("tail_id")]
    public Guid TailId { get; set; }
    [JsonPropertyName("head_id")]
    public Guid HeadId { get; set; }
    [JsonPropertyName("project_id")]
    public Guid ProjectId { get; set; }
}

public class EdgeIncomingDto : EdgeDto
{
}

public class EdgeOutgoingDto : EdgeDto
{
    [JsonPropertyName("head_node")]
    public NodeOutgoingDto HeadNode { get; set; } = new();
    [JsonPropertyName("tail_node")]
    public NodeOutgoingDto TailNode { get; set; } = new();
    [JsonPropertyName("head_issue_id")]
    public Guid HeadIssueId { get; set; }
    [JsonPropertyName("tail_issue_id")]
    public Guid TailIssueId { get; set; }
}

File: PrismaFastApi/src/dtos/edge_dtos.py
import uuid
from pydantic import BaseModel, Field
from src.dtos.node_dtos import NodeOutgoingDto


class EdgeDto(BaseModel):
    id: uuid.UUID = Field(default_factory=uuid.uuid4)
    tail_id: uuid.UUID
    head_id: uuid.UUID
    project_id: uuid.UUID


class EdgeIncomingDto(EdgeDto):
    pass


class EdgeOutgoingDto(EdgeDto):
    head_node: NodeOutgoingDto
    tail_node: NodeOutgoingDto
    head_issue_id: uuid.UUID
    tail_issue_id: uuid.UUID
